streams/streams-scala/src/test/scala/org/apache/kafka/streams/scala/KStreamTest.scala

```diff
@@ -91,6 +91,22 @@ class KStreamTest extends FlatSpec with Matchers with TestDriver {
     testDriver.close()
   }
 
+  "peek a KStream" should "side effect records" in {
```

Comment:
the test name is a bit confusing to me: peek should not have side effect on records right?

Reply:
In this test we are creating a side effect since we modify an external var.
Maybe it should be `side effect on records`?
Anyway if we change the name here we should probably change on foreach too.

Reply:
Yeah the name is a bit confusing: I thought it meant that peek / foreach can modify the records themselves, which is not correct. If it is meant for "peek / foreach can have side effects on external variables", any operators like map / filter can have that side effect as well.

Reply:
What about: should `run foreach actions on records`?

Reply:
@tedyu @joan38 that looks fine to me.